File: ConditionalSelection.js
function limitedSelection(candidates, size, minimum, maximum) {
    var final_elements = [];
    selection = groupSelection(candidates, size);
    for (var x = 0; x < selection.length; x++) {
        var state;
        var members = selection[x].split(separator);
        for (var y = 0; y < words.length; y++) {
            // get 'words[y]' frequency/count in group
            var count = 0, z = -1;
            while ((z = members.indexOf(String(words[y]), ++z)) > -1) {
                count++;
            }
            if (count >= minimum[y] && count <= maximum[y]) {
                 state = true;
            } else {
                state = false;
                break;
            }
        }
        // skip if already in net
        if (state) {
            final_elements.push(selection[x]);
        }
    }
    return final_elements;
}

File: ConditionalSelection.vb
Public Class ConditionalSelection
    Inherits Selection

    Private final_elements

    ' pick only those groups that meet occurrence conditions
    Public Function limitedSelection(candidates As String(), size As Integer, minimum As Integer(), maximum As Integer()) As List(Of String())
        final_elements = New List(Of String())
        groupSelection(candidates, size)
        For i As Integer = 0 To complete_group.Count - 1
            Dim state As Boolean = False
            For j As Integer = 0 To words.Length - 1
                ' get 'word(j)' frequency/count in group
                Dim count As Integer = -1
                Dim k As Integer = -1
                Do
                    k += 1
                    count += 1
                    k = Array.IndexOf(complete_group(i), words(j), k)
                Loop While k > -1
                ' Productive boundary check
                If count >= minimum(j) And count <= maximum(j) Then
                    state = True
                Else
                    state = False
                    Exit For
                End If
            Next j
            ' skip if already in net
            If state Then
                final_elements.Add(complete_group(i))
            End If
        Next i
        Return final_elements
    End Function

End Class
